tools/swift-inspect/Sources/swift-inspect/DumpConcurrency.swift

```diff
@@ -0,0 +1,401 @@
+      }
+      return from
+    }
+  }
+
+  func symbolicateBacktracePointer(ptr: swift_reflection_ptr_t) -> String {
+    guard let name = inspector.getSymbol(address: ptr).name else {
+      return "<\(hex: ptr)>"
+    }
+
+    return remove(from: name, upTo: " resume partial function for ")
+  }
+
+  func flagsStrings<T: BinaryInteger>(flags: T, strings: [T: String]) -> [String] {
+    return strings.sorted{ $0.key < $1.key }
+                  .filter({ ($0.key & flags) != 0})
+                  .map{ $0.value }
+  }
+
+  func flagsString<T: BinaryInteger>(flags: T, strings: [T: String]) -> String {
+    let flagStrs = flagsStrings(flags: flags, strings: strings)
+    if flagStrs.isEmpty {
+      return "0"
+    }
+
+    let flagsStr = flagStrs.joined(separator: "|")
+    return flagsStr
+  }
+
+  func decodeTaskFlags(_ info: TaskInfo) -> (
+    priority: UInt32,
+    flags: String
+  ) {
+    let priority = (info.jobFlags >> 8) & 0xff
+    let jobFlags = flagsStrings(flags: info.jobFlags, strings: [
+      1 << 24: "childTask",
+      1 << 25: "future",
+      1 << 26: "groupChildTask",
+      1 << 28: "asyncLetTask"
+    ])
+    let taskFlags = flagsStrings(flags: info.taskStatusFlags, strings: [
+      0x100: "cancelled",
+      0x200: "locked",
+      0x400: "escalated",
+      0x800: "running"
+    ])
+    let allFlags = jobFlags + taskFlags
+    let flagsStr = allFlags.isEmpty ? "0" : allFlags.joined(separator: "|")
+    return (priority, flagsStr)
+  }
+
+  func decodeActorFlags(_ flags: UInt64) -> (
+    status: String,
+    flags: String,
+    maxPriority: UInt64
+  ) {
+    let statuses: [UInt64: String] = [
+      0: "idle",
+      1: "scheduled",
+      2: "running",
+      3: "zombie-latching",
+      4: "zombie-ready-for-deallocation"
+    ]
+    let flagsString = flagsString(flags: flags, strings: [
+      1 << 3: "hasActiveInlineJob",
+      1 << 4: "isDistributedRemote"
+    ])
+
+    let status = flags & 0x7
+    let maxPriority = (flags >> 8) & 0xff
+    return (
+      status: statuses[status] ?? "unknown(\(status))",
+      flags: flagsString,
+      maxPriority: maxPriority
+    )
+  }
+
+  func dumpTasks() {
+    print("TASKS")
+
+    var lastChilds: [Bool] = []
+
+    let hierarchy = taskHierarchy()
+    for (i, (level, lastChild, task)) in hierarchy.enumerated() {
+      lastChilds.removeSubrange(level...)
+      lastChilds.append(lastChild)
+
+      let nextEntry = i < hierarchy.count - 1 ? hierarchy[i + 1] : nil
+      let prevEntry = i > 0 ? hierarchy[i - 1] : nil
+
+      let levelWillDecrease = level > (nextEntry?.level ?? -1)
+      let levelDidIncrease = level > (prevEntry?.level ?? -1)
+
+      var prefix = ""
+      for lastChild in lastChilds {
+        prefix += lastChild ? "     " : "   | "
+      }
+      prefix += "   "
+      let firstPrefix = String(prefix.dropLast(5) + (
+          level == 0 ? "   " :
+          lastChild  ? "`--" :
+                       "+--"))
```

Comment:
You didn't fancy using the box drawing characters? :-)

Reply:
Unicode in the terminal confuses and terrifies me.